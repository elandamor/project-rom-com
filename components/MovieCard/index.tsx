import { Image, StyleSheet, Text, View } from "react-native";
import { windowWidth } from "../../constants";
import slices from "../../slices";
import { Movie } from "../../types";

const cardWidth = windowWidth / 3 - 15;

export default function MovieCard({ data }: { data: Movie }) {
  const imageSource = slices[data["poster-image"].replace(".jpg", "")];
  const placeholderSource = slices["placeholderPoster"];

  return (
    <View style={styles.container}>
      <Image
        resizeMode="contain"
        source={imageSource ?? placeholderSource}
        style={styles.image}
      />
      <Text
        style={{
          color: "#fff",
          fontFamily: "TitilliumWeb_300Light",
          marginTop: 12,
        }}
        numberOfLines={1}
      >
        {data.name}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-start",
    margin: 5,
    width: cardWidth,
  },
  image: {
    height: cardWidth * 1.5,
    width: cardWidth,
  },
});
