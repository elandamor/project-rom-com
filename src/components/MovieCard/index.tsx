import { Box, Text } from "native-base";
import { Image } from "react-native";
import { windowWidth } from "../../constants";
import slices from "../../slices";
import { Movie } from "../../types";

const cardWidth = windowWidth / 3 - 15;

export default function MovieCard({ data }: { data: Movie }) {
  const imageSource = slices[data["poster-image"].replace(".jpg", "")];
  const placeholderSource = slices["placeholderPoster"];

  return (
    <Box margin={1} width={cardWidth}>
      <Image
        testID="movie-poster"
        resizeMode="contain"
        source={imageSource ?? placeholderSource}
        style={{
          height: cardWidth * 1.5,
          width: cardWidth,
        }}
      />
      <Text testID="movie-name" marginTop={3} numberOfLines={1}>
        {data.name}
      </Text>
    </Box>
  );
}
