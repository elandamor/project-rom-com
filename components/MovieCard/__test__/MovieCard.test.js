import React from "react";
import renderer from "react-test-renderer";

import MovieCard from "../index";

const testData = { name: "Rear Window", "poster-image": "poster2.jpg" };

describe("<MovieCard />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<MovieCard data={testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 2 children", () => {
    const tree = renderer.create(<MovieCard data={testData} />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});
