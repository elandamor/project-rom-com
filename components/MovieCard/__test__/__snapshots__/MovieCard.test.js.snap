// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "margin": 5,
      "width": 235,
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      {
        "height": 352.5,
        "width": 235,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      {
        "color": "#fff",
        "fontFamily": "TitilliumWeb_300Light",
        "marginTop": 12,
      }
    }
  >
    Rear Window
  </Text>
</View>
`;
